Looks like at least the RDG module is overriding one of the functions you are trying to mark `final`.
```
In file included from rdg/src/materials/AEFVMaterial.C:8:
In file included from rdg/include/materials/AEFVMaterial.h:12:
In file included from rdg/include/userobjects/SlopeLimitingBase.h:12:
In file included from rdg/include/userobjects/SlopeReconstructionBase.h:11:
rdg/include/userobjects/BCUserObject.h:50:16: error: declaration of 'threadJoin' overrides a 'final' function
  virtual void threadJoin(const UserObject & y);
```
Several of the internal apps have similar failures...

--- a/modules/rdg/include/userobjects/BCUserObject.h
+++ b/modules/rdg/include/userobjects/BCUserObject.h
@@ -47,7 +47,6 @@ public:
   virtual void initialize();
   virtual void execute();
   virtual void finalize();
-  virtual void threadJoin(const UserObject & y);
 
   /**
    * compute the ghost cell variable values

--- a/modules/rdg/include/userobjects/SlopeLimitingBase.h
+++ b/modules/rdg/include/userobjects/SlopeLimitingBase.h
@@ -27,7 +27,6 @@ public:
   SlopeLimitingBase(const InputParameters & parameters);
 
   virtual void initialize();
-  virtual void threadJoin(const UserObject & y);
   virtual void finalize();
 
   virtual void computeElement();

--- a/modules/rdg/include/userobjects/SlopeReconstructionBase.h
+++ b/modules/rdg/include/userobjects/SlopeReconstructionBase.h
@@ -28,7 +28,6 @@ public:
 
   virtual void initialize();
   virtual void finalize();
-  virtual void threadJoin(const UserObject & y);
 
   virtual void computeElement();
 

--- a/modules/rdg/src/userobjects/BCUserObject.C
+++ b/modules/rdg/src/userobjects/BCUserObject.C
@@ -31,8 +31,3 @@ void
 BCUserObject::finalize()
 {
 }
-
-void
-BCUserObject::threadJoin(const UserObject & /*y*/)
-{
-}

--- a/modules/rdg/src/userobjects/SlopeLimitingBase.C
+++ b/modules/rdg/src/userobjects/SlopeLimitingBase.C
@@ -52,14 +52,6 @@ SlopeLimitingBase::initialize()
   _lslope.clear();
 }
 
-void
-SlopeLimitingBase::threadJoin(const UserObject & y)
-{
-  const SlopeLimitingBase & pps = static_cast<const SlopeLimitingBase &>(y);
-
-  _lslope.insert(pps._lslope.begin(), pps._lslope.end());
-}
-
 const std::vector<RealGradient> &
 SlopeLimitingBase::getElementSlope(dof_id_type elementid) const
 {

--- a/modules/rdg/src/userobjects/SlopeReconstructionBase.C
+++ b/modules/rdg/src/userobjects/SlopeReconstructionBase.C
@@ -79,16 +79,6 @@ SlopeReconstructionBase::meshChanged()
   _bnd_side_area.clear();
 }
 
-void
-SlopeReconstructionBase::threadJoin(const UserObject & y)
-{
-  const SlopeReconstructionBase & pps = static_cast<const SlopeReconstructionBase &>(y);
-
-  _rslope.insert(pps._rslope.begin(), pps._rslope.end());
-
-  _avars.insert(pps._avars.begin(), pps._avars.end());
-}
-
 const std::vector<RealGradient> &
 SlopeReconstructionBase::getElementSlope(dof_id_type elementid) const
 {

--- a/test/include/postprocessors/SetupInterfaceCount.h
+++ b/test/include/postprocessors/SetupInterfaceCount.h
@@ -40,10 +40,6 @@ InputParameters validParams<InternalSideSetupInterfaceCount>();
 template <>
 InputParameters validParams<NodalSetupInterfaceCount>();
 
-namespace
-{
-}
-
 /**
  * A class for testing the number of calls to the various SetupInterface methods.
  */

